### ISSUE TITLE
Provide endpoint for AWS step functions

### ISSUE THREAD

**Issue by brettcooper** [terraform-providers/terraform-provider-aws#7735]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### I am trying to setup LocalStack with AWS Step Functions, but cannot apply my stepfunctions.tf file due to the following error: "Error: provider.aws: endpoints.0: invalid or unknown key: sfn"

It would be helpful to be able to define the AWS Step Functions endpoint.

### Potential Terraform Configuration

providers.tf:
```hcl
# Oregon
terraform {
  required_version = "0.11.11"
}

provider "aws" {

  skip_credentials_validation = true
  skip_metadata_api_check     = true
  s3_force_path_style         = true
  access_key                  = "mock_access_key"
  secret_key                  = "mock_secret_key"

  version = "1.40"
  region = "us-west-2"
  insecure = true

  # https://www.terraform.io/docs/providers/aws
  # https://github.com/localstack/localstack
  endpoints {
    # acm
    apigateway     = "https://localstack:4567"
    cloudformation = "https://localstack:4581"
    cloudwatch     = "https://localstack:4582"
    # cloudwatchevents
    # cloudwatchlogs
    # devicefarm
    dynamodb       = "https://localstack:4569"
    # ec2
    # autoscaling
    # ecr
    # ecs
    # elb
    # efs
    es             = "https://localstack:4578"
    # iam
    kinesis        = "https://localstack:4568"
    # kms
    lambda         = "https://localstack:4574"
    # r53
    # rds
    s3             = "https://localstack:4572"
    # s3control
    sns            = "https://localstack:4575"
    sqs            = "https://localstack:4576"
    # sts
    ssm            = "https://localstack:4583"
    sfn            = "https://localstack:4585"
  }
}
```

stepfunctions.tf
```hcl
resource "aws_sfn_state_machine" "SimpleDelayStateMachine" {
  name     = "SimpleDelayStateMachine"
  role_arn = "arn:aws:iam::000000000000:role/service-role/StatesExecutionRole"

  definition = <<EOF
{
   "StartAt":"start",
   "States":{
      "start":{
         "TimestampPath":"$.wait_until",
         "Next":"next",
         "Type":"Wait"
      },
      "next":{
         "End":true,
         "Resource":"arn:aws:lambda:us-west-2:000000000000:function:stepfunction-lambda-bridge",
         "Type":"Task"
      }
   }
}
EOF
}
```

### References
- https://github.com/localstack/localstack
- https://www.terraform.io/docs/providers/aws/r/sfn_state_machine.html

**Comment by bflad** [comment]
Support for configuring the endpoint has been merged and will release with version 2.5.0 of the Terraform AWS Provider likely later this week. 👍 

```hcl
provider "aws" {
  # ... potentially other provider configuration ...

  endpoints {
    # ... potentially other endpoint configuration ...
    stepfunctions  = "http://localhost:4585"
  }
}
```

**Comment by bflad** [comment]
This has been released in [version 2.5.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#250-april-05-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
